lib/v2/yoasobi-music/media.js

```diff
@@ -0,0 +1,56 @@
+const got = require('@/utils/got');
+const { parseJSONP } = require('./jsonpHelper');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const ARTIST = 'YOASOBI',
+        BASEURL = 'https://www.sonymusic.co.jp/json/v2/artist',
+        POSTFIX = 'start/0/count/-1';
+
+    const api = `${BASEURL}/${ARTIST}/media/${POSTFIX}`;
+    const officialUrl = 'https://www.yoasobi-music.jp/media';
+    const title = 'LATEST MEDIA';
+
+    const response = await got({
+        method: 'get',
+        url: api,
+    });
+
+    const data = Object.values(parseJSONP(response.data).items)
+        .reduce((p, c) => p.concat(c))
+        .sort((a, b) => new Date(b.date) - new Date(a.date))
+        .map((item) => ({
+            date: item.date,
+            weekDay: item.youbi,
+            startTime: item.startTime || null,
+            endTime: item.endTime || null,
+            tvStation: item.media || null,
+            title: item.program || item.media,
+            description: item.note,
+        }));
+
+    ctx.state.data = {
+        // the source title
+        title,
+        // the source url
+        link: officialUrl,
+        // the source description
+        description: "YOASOBI's Latest Media",
+        // iterate through all leaf objects
+        item: data.map((i) => ({
+            // the article title
+            title: i.title,
+            // the article content
+            description: art(path.join(__dirname, 'templates/media.art'), {
+                date: i.date,
+                weekDay: i.weekDay,
+                postFix: i.startTime && i.endTime && i.tvStation ? `${i.startTime} ~ ${i.endTime} ${i.tvStation}` : null,
+                description: i.description,
+            }),
+            // no pubDate
```

Comment:
`i.date` can be used as pubDate

Reply:
Actually the `i.date` refers to the air time of a tv/radio/... show, which has a different meaning with the date a post being published. Is there any side effect if I left `pubDate` param empty?

Reply:
You can leave `pubDate` empty and the feed will still work. But the user side, users cannot sort their feeds by date if it isn't provided

Reply:
OK, I see. I will use `i.date` as pubDate